apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -152,23 +154,23 @@ public void SingleSpan_Stacktrace()
         [Fact]
         public void MultipleSpans()
         {
-            Mock<ITraceConsumer> mockConsumer = new Mock<ITraceConsumer>();
+            Mock<IConsumer<TraceProto>> mockConsumer = new Mock<IConsumer<TraceProto>>();
             IManagedTracer tracer = SimpleManagedTracer.Create(mockConsumer.Object, CreateTrace());
 
             Dictionary<string, string> annotation = new Dictionary<string, string>();
             annotation.Add("annotation-key", "annotation-value");
 
             mockConsumer.Setup(c => c.Receive(
-                Match.Create<Traces>(
-                    t => t.Traces_.Count == 1 &&
-                        t.Traces_[0].Spans.Count == 5 &&
-                        IsValidSpan(t.Traces_[0].Spans[0], "child-one", t.Traces_[0].Spans[4].SpanId) &&
-                        IsValidSpan(t.Traces_[0].Spans[1], "grandchild-one", t.Traces_[0].Spans[3].SpanId, SpanKind.RpcClient) &&
-                        IsValidSpan(t.Traces_[0].Spans[2], "grandchild-two", t.Traces_[0].Spans[3].SpanId) &&
-                        TraceUtils.IsValidAnnotation(t.Traces_[0].Spans[2], annotation) &&
-                        IsValidSpan(t.Traces_[0].Spans[3], "child-two", t.Traces_[0].Spans[4].SpanId) &&
-                        !string.IsNullOrWhiteSpace(t.Traces_[0].Spans[0].Labels[Labels.StackTrace]) &&
-                        IsValidSpan(t.Traces_[0].Spans[4], "root"))));
+                Match.Create<IEnumerable<TraceProto>>(
```

Comment:
Match.Create accept a delegate, or an expression tree? If it's a delegate, I'd use a statement-bodied lambda:

```csharp
t => {
  var trace = t.Single();
  var spans = trace.Spans;
  return spans.Count == 5 &&
    IsValidSpan(spans[0], "child-one", spans[4].SpanId)
    ...
};
```

That'll reduce a lot of the duplication here.